dataplug/app/com/hubofallthings/dataplug/services/HatTokenService.scala

```diff
@@ -64,4 +65,16 @@ class HatTokenService @Inject() (protected val dbConfigProvider: DatabaseConfigP
         case _ => db.run(update).map(_ => Right(Done))
       })
   }
+
+  def delete(hat: String): Future[Done] = {
+    val q = Tables.HatToken.filter(_.hat === hat).delete
+    db.run(q).map {
```

Comment:
What will happen if there is more than one token in the table? Not an exhaustive case match

Reply:
Theoretically no. But my thinking says well even if it does happen delete all of them. The user wants to disconnect. So all the tokens should be removed.

Reply:
yes, the delete command will match every row where `_.hat === hat` and all the tokens will be removed regardless. It's the question of how our business logic handles different outcomes. 
0 deletes throws notFound error ✅ 
1 delete is expected so Done is returned
>1 delete, what happens here ❓Is it expected behaviour or unexpected behaviour? If unexpected we should raise an error.

Reply:
Yes. Agree. In my mind 1 and >1 should be the same outcome. Although the >1 should never happen because the hat field is primary key. So only one record will be there. 

Regardless of Primary Key or not though, I think even if there are more than one tokens for one HAT all of them should be deleted. Except if we want to raise an error in order to investigate it. But as a logic I think it should be the same

Reply:
> Regardless of Primary Key or not though, I think even if there are more than one tokens for one HAT all of them should be deleted.
We're not discussing whether the tokens should be removed or not here - they should, all of them. We are discussing how should the application handle the situation where the DB is found to be in an unexpected state. Logging the fact that this was the case is the very minimum we should be doing.